src/os/pl-cstack.c

```diff
@@ -940,9 +940,13 @@ sigCrashHandler(int sig)
   if ( PL_get_thread_alias(tid, &alias) )
     name = PL_atom_wchars(alias, NULL);
 
+  #pragma GCC diagnostic push
```

Comment:
I think this should be a macro defined in SWI-Prolog.h.    That at least would allow simple switch to other compilers.   In addition, some projects to not allow for `#pragma` to suppress warnings (e.g., the MSYS project)